borg/archive.py

```diff
@@ -364,7 +364,12 @@ def extract_item(self, item, restore_attrs=True, dry_run=False, stdout=False, sp
                 pos = fd.tell()
                 fd.truncate(pos)
                 fd.flush()
-                self.restore_attrs(path, item, fd=fd.fileno())
+                if sys.platform != 'win32':
+                    self.restore_attrs(path, item, fd=fd.fileno())
+                else:
+                    # File needs to be closed or timestamps are rewritten at close
+                    fd.close()
+                    self.restore_attrs(path, item)
```

Comment:
yes, you could just give it fd and it would internally compute fd.fileno() if needed / possible.